### PR TITLE
Issue 4497 Fix: Converts WorkerPostMessage into a Runnable.

### PR DESCRIPTION
**Pull request by catchmrbharath**
A lot of tests were failing when I ran `./mach test tests/wpt/web-platform-tests/workers --processes=4` and
`./mach test tests/wpt/web-platform-tests/XMLHttpRequest --processes=4` on master. So I could not completely test it. I verified that the tests that were failing were the same, before and after the change.

Is there any way to run it here?

### CODE REVIEW

File: components/script/dom/worker.rs [hunk 1]
```diff
@@ -120,3 +120,24 @@ impl<'a> WorkerMethods for JSRef<'a, Worker> {
...
+            addr: addr,
+            data: data,
+            nbytes: nbytes,
+        }
+    }
+}
+
+impl Runnable for WorkerMessageHandler {
+    fn handler(&self){
+        Worker::handle_message(self.addr.clone(), self.data, self.nbytes);
```
Comment (catchmrbharath) [inline components/script/dom/worker.rs]: Also I was not comfortable with using `clone()` here. But the borrow checker always gave me errors.

**Comment by jdm** [comment]
Looks good!

**Comment by Ms2ger** [comment]
It appears you might need to add your commit email address to your github account. Closing temporarily to get our automation running again.

**Comment by catchmrbharath** [comment]
If its already not merged, I will change the email address.

**Comment by catchmrbharath** [comment]
@jdm I seem to have messed up while rebasing. Please *do not merge* as of now. Sorry for the mess up.

**Comment by catchmrbharath** [comment]
@jdm I have fixed it. Can you approve the pull request again?

**Pull request merged**